SPEC.md

```diff
@@ -4586,31 +4618,58 @@ Example output:
+* Alias: `maxMemory`
 
 Specifies the maximum memory provisioned for a task. The value of this hint has the same specification as [`requirements.memory`](#memory).
 
-##### `shortTask`
+##### ✨ `volumes`
 
-* Allowed type: `Boolean`
+* Accepted types: `Map[String, String]`
 
-A `Boolean` value for which `true` indicates that that this task is not expected to take long to execute. The execution engine can interpret this as permission to attempt to optimize the execution of the task - e.g., by batching together multiple `shortTask`s, or by using the cost-optimized instance types that many cloud vendors provide, e.g., `preemptible` instances on `GCP` and `spot` instances on `AWS`. "Short" is a bit ambiguous, but should generally be interpreted as << 24h.
+Specifies attributes for any [disk mount points](#disks) that are required. The value of this hint is a `Map` with the key being the mount point and the value being a `String` with the volume specification. The values "HDD" and "SSD" should be recognized to indicate that a specific class of hardware is being requested. Volume specifications are left intentionally vague as they are primarily intented to be used in the context of a specific [compute environment](#compute-environments).
 
-##### `localizationOptional`
+##### ✨ `gpus` and ✨ `fpgas`
 
-* Allowed type: `Boolean`
+* Accepted types:
+    * `Int`: Minimum number of accelerators being requested.
+    * `String`: Specification for a type of accelerator being requested, e.g., manufacturer or model name.
+
+Specifies attributes for any [hardware accelerators](#hardware-accelerators-gpu-and-✨-fpga) that are requested. Accelerator specifications are left intentionally vague as they are primarily intented to be used in the context of a specific [compute environment](#✨-compute-environments).
+
+##### `short_task`
+
+* Accepted types: `Boolean`
+* Default value: `false`
+
+A `Boolean` value for which `true` indicates that that this task is not expected to take long to execute. The execution engine can interpret this as permission to attempt to optimize the execution of the task. For example, the engine may batch together multiple `short_task`s, or it may use the cost-optimized instance types that many cloud vendors provide, e.g., `preemptible` instances on `GCP` and `spot` instances on `AWS`.
+
+##### `localization_optional`
+
+* Accepted types: `Boolean`
+* Default value: `false`
+* Alias: `localizationOptional`
 
 A `Boolean` value, for which `true` indicates that, if possible, any `File` inputs for this task should not be (immediately) localized. For example, a task that processes its input file once in linear fashion could have that input streamed (e.g., using a `fifo`) rather than requiring the input file to be fully localized prior to execution. This directive must not have any impact on the success or failure of a task (i.e., a task must run the same with or without localization).
```

Comment:
> This directive must not have any impact on the success or failure of a task (i.e., a task must run the same with or without localization).

I think this means that the script of the task (and the applications used by the task) must be able to work in either case (e.g. they can use a local file path or a s3:// or gs:// style path). Or should this be interpreted to mean that the engine must be able to detect the capabilities of the task and called applications and do the right thing? Maybe we need to say who is responsible for ensuring this directive doesn't impact success (the WDL workflow author or the workflow engine).

Reply:
This hint is saying that the task is able to handle a sequential read-only input file, i.e., it's not going to perform random access. Thus, the engine is free to e.g. provide a FIFO rather than a real file, and stream the input to that FIFO.